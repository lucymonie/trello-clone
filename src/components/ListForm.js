import React, { Component } from 'react';
import SaveCancel from './SaveCancel';

class ListForm extends Component {
  constructor() {
    super();
    this.state = {}
  }

  render() {
    const { onClickSave, onClickCancel, onFocus, onChange, textEntered, showSave } = this.props;
    return (
    <div className="list-wrapper">
      <div className="new-list indigo lighten-5">
        <input className="new-list-input"
          onFocus={ onFocus }
          onChange={ (e) => onChange(e.target.value) }
          placeholder="Add a list..."
          value={ textEntered } />
        {showSave &&
        <SaveCancel 
          onClickSave={ onClickSave }
          onClickCancel={ onClickCancel } />
        }
      </div>
    </div>
    )
  }
}

export default ListForm;
