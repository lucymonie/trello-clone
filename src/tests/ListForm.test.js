import React from 'react';
import { shallow } from 'enzyme';
import ListForm from '../components/ListForm';


describe("ListForm", () => {
  it("renders consistently", () => {
    const component = shallow(<ListForm showSave={false} />);
    expect(component).toMatchSnapshot();
  });

  it("should render textEntered in the input field", () => {
    const component = shallow(<ListForm showSave={false} textEntered="" />);
    const inputField = component.find("input");
    expect(inputField.text()).toBe("");
  });

  it("should render SaveCancel if showSave is true", () => {
    const component = shallow(<ListForm showSave={true} textEntered="" />);
    const saveCancel = component.find("SaveCancel");
    expect(component).toMatchSnapshot();
  });

  describe("when focusing on the input field", () => {
    it("triggers the onFocus", () => {
      const mockEvent = jest.fn();
      const component = shallow(<ListForm showSave={false} textEntered="" onFocus={mockEvent} />);
      const inputField = component.find("input").simulate("focus");
      expect(mockEvent.mock.calls.length).toEqual(1);
    });
  });
});
