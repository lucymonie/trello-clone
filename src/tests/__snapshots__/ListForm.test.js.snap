// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListForm renders consistently 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListForm
    showSave={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="new-list indigo lighten-5"
>
        <input
                className="new-list-input"
                onChange={[Function]}
                onFocus={undefined}
                placeholder="Add a list..."
                value={undefined}
        />
</div>,
      "className": "list-wrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="new-list-input"
            onChange={[Function]}
            onFocus={undefined}
            placeholder="Add a list..."
            value={undefined}
/>,
          false,
        ],
        "className": "new-list indigo lighten-5",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "new-list-input",
            "onChange": [Function],
            "onFocus": undefined,
            "placeholder": "Add a list...",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        false,
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="new-list indigo lighten-5"
>
          <input
                    className="new-list-input"
                    onChange={[Function]}
                    onFocus={undefined}
                    placeholder="Add a list..."
                    value={undefined}
          />
</div>,
        "className": "list-wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="new-list-input"
              onChange={[Function]}
              onFocus={undefined}
              placeholder="Add a list..."
              value={undefined}
/>,
            false,
          ],
          "className": "new-list indigo lighten-5",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "new-list-input",
              "onChange": [Function],
              "onFocus": undefined,
              "placeholder": "Add a list...",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          false,
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ListForm should render SaveCancel if showSave is true 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListForm
    showSave={true}
    textEntered=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="new-list indigo lighten-5"
>
        <input
                className="new-list-input"
                onChange={[Function]}
                onFocus={undefined}
                placeholder="Add a list..."
                value=""
        />
        <SaveCancel
                onClickCancel={undefined}
                onClickSave={undefined}
        />
</div>,
      "className": "list-wrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="new-list-input"
            onChange={[Function]}
            onFocus={undefined}
            placeholder="Add a list..."
            value=""
/>,
          <SaveCancel
            onClickCancel={undefined}
            onClickSave={undefined}
/>,
        ],
        "className": "new-list indigo lighten-5",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "new-list-input",
            "onChange": [Function],
            "onFocus": undefined,
            "placeholder": "Add a list...",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onClickCancel": undefined,
            "onClickSave": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="new-list indigo lighten-5"
>
          <input
                    className="new-list-input"
                    onChange={[Function]}
                    onFocus={undefined}
                    placeholder="Add a list..."
                    value=""
          />
          <SaveCancel
                    onClickCancel={undefined}
                    onClickSave={undefined}
          />
</div>,
        "className": "list-wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="new-list-input"
              onChange={[Function]}
              onFocus={undefined}
              placeholder="Add a list..."
              value=""
/>,
            <SaveCancel
              onClickCancel={undefined}
              onClickSave={undefined}
/>,
          ],
          "className": "new-list indigo lighten-5",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "new-list-input",
              "onChange": [Function],
              "onFocus": undefined,
              "placeholder": "Add a list...",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClickCancel": undefined,
              "onClickSave": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
